packages/nemo-evaluator-launcher/src/nemo_evaluator_launcher/common/mapping.py

```diff
@@ -145,83 +73,332 @@ def _process_mapping(mapping_toml: dict) -> dict:
                 }
+                # Only add container if it exists
+                if container:
+                    mapping[key]["container"] = container
+
+                # Validate task_data keys before updating
                 for task_data_key in task_data.keys():
                     if task_data_key in mapping[key]:
                         raise KeyError(
                             f"{repr(task_data_key)} is not allowed as key under {repr(key)} in the mapping"
                         )
+                    # Validate that task_data values are valid types (basic check)
+                    if task_data_key not in ("description", "type") and not isinstance(
+                        task_data[task_data_key],
+                        (str, int, float, bool, dict, list, type(None)),
+                    ):
+                        logger.warning(
+                            "Unexpected value type in task_data",
+                            harness_name=harness_name,
+                            task_name=task_name,
+                            key=task_data_key,
+                            value_type=type(task_data[task_data_key]).__name__,
+                        )
+
                 mapping[key].update(task_data)
     return mapping
 
 
+def _extract_tasks_from_framework_yml(
+    framework_yml_content: str, harness_name: str, container: str
+) -> dict[tuple[str, str], dict]:
+    """Extract tasks from framework.yml content and return as mapping entries.
+
+    Args:
+        framework_yml_content: YAML content from framework.yml file
+        harness_name: Name of the harness
+        container: Container image string
+
+    Returns:
+        Dictionary mapping (harness_name, task_name) to task configuration
+    """
+    tasks = {}
+    try:
+        framework_data = yaml.safe_load(framework_yml_content)
+        if not framework_data or "evaluations" not in framework_data:
+            logger.warning(
+                "No evaluations found in framework.yml",
+                harness=harness_name,
+                container=container,
+            )
+            return tasks
+
+        evaluations = framework_data.get("evaluations", [])
+        for eval_config in evaluations:
+            task_name = eval_config.get("name")
+            description = eval_config.get("description", "")
+
+            if not task_name:
+                continue
+
+            # Extract endpoint types from the evaluation config
+            defaults = eval_config.get("defaults", {})
+            config = defaults.get("config", {})
+            supported_endpoint_types = config.get("supported_endpoint_types", ["chat"])
+            task_type = config.get("type", "")  # Extract type from defaults.config.type
+
+            # Use first endpoint type (mapping key is (harness, task), so one entry per task)
+            endpoint_type = (
+                supported_endpoint_types[0] if supported_endpoint_types else "chat"
```

Comment:
please add a FIXME note here: we should have one endpoint type per task. but for this to happen we need to first design how we want to define endpoint types and flavlors. cc @prokotg